### PR TITLE
feat: Integrate new pubsubx with franz-go

### PR DESCRIPTION
**Pull request by xWiiLLz**
This PR brings in the new revision of our `pubsubx` package. We no longer rely on `watermill`, and are now using `franz-go` as our official kafka implementation.

This PR also introduces the basic tracing capabilities for our `kgox` package.

### CODE REVIEW

File: .github/workflows/ci-kafka.yml [hunk 1]
```diff
@@ -17,11 +17,12 @@ jobs:
         with:
           go-version: '^1.21.1'
       - run: cat .env >> $GITHUB_ENV || true
-      - run: docker compose -f docker-compose.kafkax.yml up -d
+      - run: docker compose -f docker-compose.kgox.yml up -d
       - run: go run github.com/ThreeDotsLabs/wait-for@latest localhost:9091 localhost:9092 localhost:9093 localhost:9094 localhost:9095
```
Comment (Sansinj) [inline .github/workflows/ci-kafka.yml:21]: Could we remove this ?

File: logrusx/log_fields.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package logrusx
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+func NewLogFields(kvs ...attribute.KeyValue) logrus.Fields {
```
Comment (Sansinj) [inline logrusx/log_fields.go:10]: I would a quick comment to explain why we need this.

File: pubsubx/inmemory.go [hunk 3]
```diff
@@ -2,47 +2,122 @@ package pubsubx
...
+// Subscribe implements Subscriber.
+func (m *memorySubscriber) Subscribe(ctx context.Context, topicHandlers Handlers) error {
+	m.m.mu.Lock()
+	defer m.m.mu.Unlock()
+	for _, topic := range m.topics {
+		handler, exists := topicHandlers[topic]
+		if !exists {
+			return errorx.InvalidArgumentErrorf("missing handler for topic %s", topic)
+		} else if handler == nil {
+			return errorx.InvalidArgumentErrorf("nil handler for topic %s", topic)
```
Comment (Sansinj) [inline pubsubx/inmemory.go:116]: Internal Error ?

Comment (Sansinj) [inline pubsubx/inmemory.go:116]: Internal Error

File: .github/workflows/ci-kafka.yml (hunk 1 above)
Comment (xWiiLLz) [reply .github/workflows/ci-kafka.yml:21]: Do you mean the test at all or just this line? If the latter, I believe we still need to wait for the kafka brokers to all be ready. I'd keep as is IMO

File: pubsubx/kgox/pubsub.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+		return c, nil
+	}
+	p.mu.RUnlock()
+
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	o := &pubsubx.SubscriberOptions{
+		MaxBatchSize: 100,
+	}
```
Comment (Sansinj) [inline pubsubx/kgox/pubsub.go:96]:
```suggestion
	o := pubsubx.NewDefaultSubscriberOptions()

```

File: .github/workflows/ci-kafka.yml (hunk 1 above)
Comment (Sansinj) [reply .github/workflows/ci-kafka.yml:21]: Yes NVM

File: pubsubx/kgox/pubsub.go (hunk 4 above)
Comment (xWiiLLz) [reply pubsubx/kgox/pubsub.go]: Good catch

**Review by Sansinj — APPROVED** [review]

**Pull request merged**